packages/modeling/src/geometries/types.d.ts

```diff
@@ -2,11 +2,16 @@ import Geom2 from './geom2/type'
 import Geom3 from './geom3/type'
 import Poly3 from './poly3/type'
 import Path2 from './path2/type'
+import { RGB, RGBA } from '../colors'
 
 // NOTE (@ahdinosaur): Poly2 is not included as not well-supported.
 // see https://github.com/jscad/OpenJSCAD.org/pull/726#issuecomment-724575265
 export type Geometry = Geom2 | Geom3 | Poly3 | Path2
 
+export type Colored = {
```

Comment:
TypeScript requires all properties to be defined, but those that are not required can be defined optionally.
Functions that use only certain properties can be typed in a duck-typed way using generics.
It might even be possible to create a base type for Geometry with the properties `color?` and `name?`.

Reply:
Ok. Let’s create a (cough) base type.